index.js

```diff
@@ -26,10 +26,27 @@ function componentWillReceiveProps(nextProps) {
   }
 }
 
+function componentWillUpdate(nextProps, nextState) {
+  try {
+    var prevProps = this.props;
+    var prevState = this.state;
+    this.props = nextProps;
+    this.state = nextState;
```

Comment:
Won't direct assignments to these warn?

Reply:
No. React sets/updates this values in a similar way before calling lifecycles.